src/main/java/com/limechain/trie/structure/Vacant.java

```diff
@@ -419,52 +430,57 @@ private Two(TrieStructure<T> trieStructure,
             /**
              * Insert the new storage node (and the intermediate branch node needed for this case)
              * Branch node's user data defaults to null.
+             *
              * @param storageUserData the userData held by the storage node
              * @return a StorageNodeHandle to the freshly inserted storage node
              */
-            public StorageNodeHandle<T> insert(T storageUserData) {
-                return this.insert(storageUserData, null);
+            @Override
+            public StorageNodeHandle<T> insert(T storageUserData, StateVersion stateVersion) {
+                return this.insert(storageUserData, null, stateVersion);
             }
 
             /**
              * Insert the new storage node (and the intermediate branch node needed for this case)
+             *
              * @param storageUserData the userData held by the storage node
-             * @param branchUserData the userData held by the branch node
+             * @param branchUserData  the userData held by the branch node
              * @return a StorageNodeHandle to the freshly inserted storage node
              */
             @SuppressWarnings("unused")
-            public StorageNodeHandle<T> insert(T storageUserData, T branchUserData) {
+            public StorageNodeHandle<T> insert(T storageUserData, T branchUserData, StateVersion stateVersion) {
                 int newBranchNodePartialKeyLen = this.branchPartialKey.size();
 
                 assert 1 == Arrays.stream(this.branchChildrenIndices).filter(Objects::nonNull).count()
-                    : "The branch node we're about to insert must have exactly one child " +
-                        "(the node that previously existed before this insertion)";
+                        : "The branch node we're about to insert must have exactly one child " +
+                          "(the node that previously existed before this insertion)";
 
                 // Insert the intermediate branch node
                 int newBranchNodeIndex = this.trieStructure.nodes.add(new TrieNode<>(
-                    this.branchParent,
-                    this.branchPartialKey,
-                    // NOTE: Crucial mutability moment, this has to be a copy of the array.
-                    //  I'll never get those three hours of my life back :)
-                    //  Manage your ownership, kids.
-                    ArrayUtils.clone(this.branchChildrenIndices),
-                    false,
-                    branchUserData
+                        this.branchParent,
+                        this.branchPartialKey,
+                        // NOTE: Crucial mutability moment, this has to be a copy of the array.
+                        //  I'll never get those three hours of my life back :)
+                        //  Manage your ownership, kids.
```

Comment:
I know it's not written by you but you can delete the second and 3rd row of that comment.